diff --git a/datapath/datapath.c b/datapath/datapath.c
index deba048..0c34a7e 100644
--- a/datapath/datapath.c
+++ b/datapath/datapath.c
@@ -1823,6 +1823,12 @@ static void set_desc(void)
                 strlcpy(serial_num, serial, sizeof(serial_num));
 }
 
+#ifdef CONFIG_BRIDGE
+extern struct notifier_block br_device_notifier;
+int dp_device_event(struct notifier_block *, unsigned long,  void *);
+int (*saved_br_device_notifier_call)(struct notifier_block *, unsigned long,  void *);
+#endif
+
 static int __init dp_init(void)
 {
 	int err;
@@ -1851,6 +1857,10 @@ static int __init dp_init(void)
 	if (br_handle_frame_hook)
 		printk("openflow: hijacking bridge hook\n");
 	br_handle_frame_hook = dp_frame_hook;
+#ifdef CONFIG_BRIDGE
+	saved_br_device_notifier_call = br_device_notifier.notifier_call;
+	br_device_notifier.notifier_call = dp_device_event;
+#endif
 
 	return 0;
 
@@ -1870,6 +1880,9 @@ static void dp_cleanup(void)
 	unregister_netdevice_notifier(&dp_device_notifier);
 	flow_exit();
 	br_handle_frame_hook = NULL;
+#ifdef CONFIG_BRIDGE
+	br_device_notifier.notifier_call = saved_br_device_notifier_call;
+#endif
 }
 
 module_init(dp_init);
diff --git a/datapath/dp_notify.c b/datapath/dp_notify.c
index 9bf491d..f22f91c 100644
--- a/datapath/dp_notify.c
+++ b/datapath/dp_notify.c
@@ -10,9 +10,8 @@
 
 #include "datapath.h"
 
-
-static int dp_device_event(struct notifier_block *unused, unsigned long event, 
-		void *ptr) 
+int
+dp_device_event(struct notifier_block *unused, unsigned long event, void *ptr)
 {
 	struct net_device *dev = ptr;
 	struct net_bridge_port *p = dev->br_port;
